### DISCUSSION TITLE
Pass data to subscribe callback

### DISCUSSION THREAD

**Discussion by helander** [pocketbase/pocketbase#2048] [Q&A]
Is it possible to pass application data to the callback function of the subscribe operation ?

E.g.
`pb.collection('example').subscribe('*', function (e) {
    console.log(e.myownspecialdata);
});`

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand your use case, but you can attach to the [`OnRealtimeBeforeMessageSend` event hook](https://pocketbase.io/docs/event-hooks/#realtime-api-hooks) and overwrite the `e.Message` encoded json.

**Reply by helander** [reply]
My ”example” was from the Javascript SDK, i.e. my question was related to a Javascript client. 
The callback function actions are dependent on the “context” of the subscribe call, so I need to pass context information that are available in the callback function.

**Reply by ganigeorgiev** [reply]
I'm sorry, I still don't understand your use case and what you are trying to do.

What is `myownspecialdata` and from where it comes?
Is it related to the realtime event somehow or it is just some local variable that you want to use inside the callback?

If you just want to "encapsulate" some local variable as part of the subscribe callback, you can try creating a wrapper function:

```js
function wrapSubscribeFunc(extraData) {
    return (e) => {
        console.log(e.action, extraData);
    }
}

pb.collection('example').subscribe('*', wrapSubscribeFunc(yourExtraData));
```

If that's not what you are trying to do, please elaborate and provide a little more details about your use case.

**Reply by helander** [reply]
I think that a wrapper function might work for my use case, thanks.
